packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -338,8 +361,27 @@ class CupertinoSlidingSegmentedControl<T extends Object> extends StatefulWidget
   /// The map must have more than one entry.
   final Map<T, Widget> children;
 
+  /// The set of identifying keys that correspond to the segments that should be
+  /// disabled.
+  ///
+  /// Disabled children cannot be highlighted and don't show any animation
+  /// when they are tapped or long pressed. So if this set contains [groupValue],
+  /// there is no segment getting highlighted until it is switched to an enabled
+  /// segment.
+  ///
+  /// [onValueChanged] will not be called if the segment is disabled. However, if
```

Comment:
The [onValueChanged] callback will not be called if the currently selected segment becomes disabled,